packages/view-types/scatterplot/src/shared-spatial-scatterplot/AbstractSpatialOrScatterplot.js

```diff
@@ -19,12 +19,19 @@ export default class AbstractSpatialOrScatterplot extends PureComponent {
     };
 
     this.viewport = null;
-
+    this.originalViewState = null;
     this.onViewStateChange = this.onViewStateChange.bind(this);
     this.onInitializeViewInfo = this.onInitializeViewInfo.bind(this);
     this.onWebGLInitialized = this.onWebGLInitialized.bind(this);
     this.onToolChange = this.onToolChange.bind(this);
     this.onHover = this.onHover.bind(this);
+
+    console.log("In AbstractSpatialOrScatterplot");
+  }
+
+  recenter(){
+    console.log("I am really recentering now: ", this.originalViewState);
+    this.onViewStateChange(this.originalViewState);
   }
```

Comment:
define the custom function for the zoom button, which is to bring back the original zoom and x and y positions